src/components/timers/Timer.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState, useEffect } from 'react';
+import DisplayRounds from '../displayRounds/DisplayRounds';
+import DisplayTime from '../displayTime/DisplayTime';
+import Button from '../button/Button';
+
+type TimerType = 'stopwatch' | 'countdown' | 'xy' | 'tabata';
+
+interface TimerProps {
+    type: TimerType;
+    startTime?: number; // Used for countdown
+    workTime?: number; // Used for tabata
+    restTime?: number; // Used for tabata
+    roundTime?: number; // Used for xy
+    rounds?: number; // Used for xy and tabata
+}
+
+const Timer: React.FC<TimerProps> = ({ type, startTime = 0, workTime = 20, restTime = 10, roundTime = 60, rounds = 1 }) => {
+    const [time, setTime] = useState(type === 'countdown' ? startTime : type === 'xy' ? roundTime : 0);
+    const [isActive, setIsActive] = useState(false);
+    const [isPaused, setIsPaused] = useState(false);
+    const [isWorkInterval, setIsWorkInterval] = useState(true); // Start with a work interval for Tabata
+    const [currentRound, setCurrentRound] = useState(1);
+
+    // Timer effect
+    useEffect(() => {
+        let timer: NodeJS.Timeout | null = null;
+
+        if (isActive && !isPaused) {
+            timer = setInterval(() => {
```

Comment:
you are clearing and setting a new interval every time `time` changes, so there's no point in using an interval. use `setTimeout` instead, which only fires once and has less risk of creating a memory leak if you forget to clear it.